make/test.mk

```diff
@@ -73,82 +73,51 @@ HOST_NS := host-operator-$(shell date +'%s')
-	@oc logs deployment.apps/host-operator --namespace $(HOST_NS)
-	@echo "====================================================================================="
-	@echo "================================ Member cluster logs ================================"
-	@echo "====================================================================================="
-	@oc logs deployment.apps/member-operator --namespace $(MEMBER_NS)
-	@echo "====================================================================================="
-
-.PHONY: e2e-setup
-e2e-setup: is-minishift
-	oc new-project $(HOST_NS) --display-name e2e-tests
-	oc apply -f ./deploy/service_account.yaml
-	oc apply -f ./deploy/role.yaml
-	oc apply -f ./deploy/role_binding.yaml
-	oc apply -f ./deploy/cluster_role.yaml
-	sed -e 's|REPLACE_NAMESPACE|${HOST_NS}|g' ./deploy/cluster_role_binding.yaml  | oc apply -f -
-	oc apply -f deploy/crds
-	sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ./deploy/operator.yaml  | oc apply -f -
+.PHONY: test-e2e-local
+test-e2e-local:
+	$(MAKE) test-e2e E2E_REPO_PATH=../toolchain-e2e
 
-.PHONY: is-minishift
-is-minishift:
-ifeq ($(OPENSHIFT_BUILD_NAMESPACE),)
-	$(info logging as system:admin")
-	$(shell echo "oc login -u system:admin")
-	$(eval IMAGE_NAME := docker.io/${GO_PACKAGE_ORG_NAME}/${GO_PACKAGE_REPO_NAME}:${GIT_COMMIT_ID_SHORT})
-	$(shell echo "make docker-image")
-else
-	$(eval IMAGE_NAME := registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:host-operator)
+.PHONY: test-e2e
+test-e2e: get-e2e-repo
+	# run the e2e test via toolchain-e2e repo
+	$(MAKE) -C ${E2E_REPO_PATH} test-e2e HOST_REPO_PATH=${PWD}
+
+.PHONY: get-e2e-repo
+get-e2e-repo:
+ifeq ($(E2E_REPO_PATH),"")
+	# set e2e repo path to tmp directory
+	$(eval E2E_REPO_PATH = /tmp/toolchain-e2e)
+	# delete to have clear environment
+	rm -rf ${E2E_REPO_PATH}
+	# clone
+	git clone https://github.com/codeready-toolchain/toolchain-e2e.git --depth 1 ${E2E_REPO_PATH}
+    ifneq ($(CLONEREFS_OPTIONS),)
+		@echo "using author link ${AUTHOR_LINK}"
+		@echo "using pull sha ${PULL_SHA}"
+		# get branch ref of the fork the PR was created from
+		$(eval BRANCH_REF := $(shell curl ${AUTHOR_LINK}/host-operator.git/info/refs?service=git-upload-pack --output - /dev/null 2>&1 | grep -a ${PULL_SHA} | awk '{print $$2}'))
+		@echo "detected branch ref ${BRANCH_REF}"
+		# check if a branch with the same ref exists in the user's fork of toolchain-e2e repo
+		$(eval REMOTE_E2E_BRANCH := $(shell curl ${AUTHOR_LINK}/toolchain-e2e.git/info/refs?service=git-upload-pack --output - /dev/null 2>&1 | grep -a ${BRANCH_REF} | awk '{print $$2}'))
+		@echo "branch ref of the user's fork: \"${REMOTE_E2E_BRANCH}\" - if empty then not found"
+		# check if the branch with the same name exists, if so then merge it with master and use the merge branch, if not then use master
+		if [[ -n "${REMOTE_E2E_BRANCH}" ]]; then \
+			# retrieve the branch name; \
+			BRANCH_NAME=`echo ${BRANCH_REF} | awk -F'/' '{print $$3}'`; \
```

Comment:
You want to remove `\` from all the comments in this block, don't you? :) Unless you want to keep entire `then` block as a big comment.

Reply:
no, the slash is needed there, otherwise, the next line is evaluated as a new Makefile line - not as part of the "one-line" bash command.

Reply:
Ah right. But it means you can’t use `#` as comments here at all. Unless I’m missing something.

Reply:
it's possible - the back-slack just escapes a new line. It works, in the same way as you would write a normal bash script - you can put comments there as well ;-) 
just paste this into your terminal and you will see that it's working:
```bash
if [[ true ]]; then \
echo "before comment"; \
# comment \
echo "after comment"; \
fi;
```
in bash, it actually works in the same way also without the back-slack behind the comment:
```bash
if [[ true ]]; then \
echo "before comment"; \
# comment 
echo "after comment"; \
fi;
```
but to be properly parsed by Makefile, I need to have it there in my case. 
The only thing I can remove is the semicolon at the end of the comments, the back-slash has to stay there

Reply:
but if the comments are confusing, I can remove them completely - I just thought that it would be good to have the steps commented :-)

Reply:
No, no, keep the comments. They are useful.
I just missed `;` in the end of the comments which separates the comment from the following code in the same line. Bash was never my strong side :)

So, all looks good here.

Reply:
Oh, `;` are actually not needed. Bash seems to be smart enough to convert
```
# some comment \
some code;
```
to `# some comment; some code;`

Anyway, the code looks good.